### DISCUSSION TITLE
React.memo, useCallback, useMemo 등의 성능최적화 적용의 판단기준은 무엇일까

### DISCUSSION THREAD

**Discussion by zereight** [woowacourse/react-payments#43] [토론장]
React.memo, useCallback, useMemo 등의 성능최적화 적용의 판단기준은 무엇일까

그냥 느낌적으로 느려짐이 느껴질때?

일단 적용해보고 개선됨이 체감될때?

수치상으로 성능이 저하됨이 나타날때? 그럼 그 성능 측정도구와 방법은?

**Comment by Puterism** [comment]
느낌적으로 느려짐이 느껴질 때 -> 이 때는 정말 최적화 적용이 필요할 것 같습니다.
일단 적용해보고 개선됨이 체감될 때 -> 일단 적용하지 않는 게 좋을 것 같아요. 최적화도 또 다른 비용이라고 생각합니다.
수치상으로 성능 저하가 나타날 때 -> 주기적인 성능 측정을 진행하는 것도 좋은 방법 같아요.
